src/registries/yarn-registry.js

```diff
@@ -56,7 +56,15 @@ export default class YarnRegistry extends NpmRegistry {
   homeConfig: Object;
 
   getOption(key: string): mixed {
-    return this.config[key] || this.registries.npm.getOption(npmMap[key] || key);
+    let val = this.config[key] || this.registries.npm.getOption(npmMap[key]);
+
+    // if we have no yarn option for this or have used a default then use the npm
+    // value if it exists
+    if (!val || val === DEFAULTS[key]) {
```

Comment:
With this code, if the _value_ is the same as the default, it will favor the npm config even if the user explicitly set it. Shouldn't an explicit setting still take precedence over the npm config value?